### ISSUE TITLE
Heatmaps should have a toggle to filter out internal users

### ISSUE THREAD

**Issue by marandaneto** [PostHog/posthog#21604]
## Is your feature request related to a problem?

[Heatmaps](https://posthog.com/docs/toolbar/heatmaps)

## Describe the solution you'd like

A toggle to filter out internal users similar to [this](https://posthog.com/tutorials/filter-internal-users)

## Additional context

https://github.com/PostHog/posthog/blob/master/posthog/api/element.py

### Debug info

```shell
- [ ] PostHog Cloud, Debug information: [please copy/paste from https://us.posthog.com/settings/project-details#variables]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see [`Sunsetting Kubernetes support`](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]
```

**Comment by pauldambra** [comment]
see also https://posthoghelp.zendesk.com/agent/#/tickets/21617

to filter by internal users we'd need to support _at least_ person and event properties in the query code, at that point we may as well simply allow event and person filtering

**Comment by kendagriff** [comment]
Yes, the ability to filter test users from heat maps would be super helpful.

**Comment by pauldambra** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/22153 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/23822)

**Comment by bendbradley** [comment]
A prospective customer asked for the ability to filter Heatmaps by person properties, cohorts and groups.

**Comment by pauldambra** [comment]
will punt more complete filtering to #21531

but internal user filtering should work for most uses from https://github.com/PostHog/posthog/pull/30573

**Comment by pauldambra** [comment]
they do now - in the in-app heatmaps beta https://us.posthog.com/#panel=feature-previews%3Aheatmaps-ui
